### ISSUE TITLE
option for "infinity-math"

### ISSUE THREAD

**Issue by jxwuyi** [pytorch/pytorch#1817]
I'm using Gumbel-Softmax trick, that is I need to compute:
`softmax(-log(-log(uniform(0, 1))))`
In Tenserflow, 
running `tf.nn.softmax(-tf.log(-tf.log(tf.uniform(...))))` is fine.
But in Pytorch,
if you run `F.softmax(-torch.log(-torch.log(torch.rand())))`
You will get "nan".
I think the reason is: rand() will sometimes return exactly 0; while in tensorflow exp(-inf) will produce 0 while in pytorch "nan" will be thrown.
Any option for pytorch to allow "infinity-math"???

**Comment by apaszke** [comment]
```python
torch.Tensor([float('-inf')]).exp()
```
gives me 0

**Comment by soumith** [comment]
as Bharath pointed out internally:

> just clamp the value?
F.softmax(-torch.log(-torch.log(torch.rand().clamp(min=1e-5))))
